### PR TITLE
Implement visual diff

### PR DESCRIPTION
**Pull request by FluorescentHallucinogen**
This pull request adds a visual diff between two GraphQL schemas.

It should significantly improve the developer experience (DX) as compared to plaintext JSON diff.

Added fields are marked in green color, deleted in red and modified in yellow. Every change is explained in tooltip on affected field hover.

graphql-visual-diff

### CODE REVIEW

File: packages/commands/diff/package.json
Comment (FluorescentHallucinogen) [inline packages/commands/diff/package.json:38]: @kamilkisiela This line could be removed when [this bug](https://github.com/ardatan/graphql-tools/issues/2051#issuecomment-699916227) is fixed.

@ardatan cc.

File: packages/commands/diff/src/static/index.html
Comment (FluorescentHallucinogen) [inline packages/commands/diff/src/static/index.html:1]: @kamilkisiela Need your help. How to include (copy) this file into build?

File: packages/commands/diff/src/index.ts
Comment (FluorescentHallucinogen) [inline packages/commands/diff/src/index.ts:266]: @kamilkisiela Need your help. I've commented out this line, because it stops the Express. Is it ok? Is there a better way to fix (avoid) this? This pull request allows edits by maintainers. Feel free to help me by pushing your fixes.

Comment (kamilkisiela) [reply packages/commands/diff/src/index.ts:266]: @FluorescentHallucinogen could we hide it behind a `--visual` flag to not introduce a breaking change?

**Review by kamilkisiela — CHANGES_REQUESTED** [review]

File: packages/commands/diff/src/static/index.html
Comment (kamilkisiela) [reply packages/commands/diff/src/static/index.html:1]: We would have to adjust [`bob-the-bundler`](https://github.com/kamilkisiela/bob) but I'll do it after merging this PR.

File: packages/commands/diff/src/index.ts
Comment (FluorescentHallucinogen) [reply packages/commands/diff/src/index.ts]: @kamilkisiela Done.

File: packages/commands/diff/package.json
Comment (kamilkisiela) [reply packages/commands/diff/package.json:38]: @FluorescentHallucinogen looks good but let's wait for the fix @ardatan

**Comment by FluorescentHallucinogen** [comment]
@kamilkisiela @dotansimha @ardatan @Urigo

Also visual diff from this pull request could be a very useful addition to GraphQL Hive:

graphql-hive

All that is needed is to execute the next GraphQL query to https://app.graphql-hive.com/api/proxy endpoint

```graphql
query compare($organization: ID!, $project: ID!, $target: ID!, $version: ID!) {
  schemaCompareToPrevious(
    selector: {organization: $organization, project: $project, target: $target, version: $version}
  ) {
    __typename
    ...SchemaCompareResultFields
    ...SchemaCompareErrorFields
  }
  schemaVersion(
    selector: {organization: $organization, project: $project, target: $target, version: $version}
  ) {
    __typename
    ...SchemaVersionFields
  }
}

fragment SchemaCompareResultFields on SchemaCompareResult {
  initial
  changes {
    total
    nodes {
      ...SchemaChangeFields
      __typename
    }
    __typename
  }
  diff {
    before
    after
    __typename
  }
}

fragment SchemaChangeFields on SchemaChange {
  path
  message
  criticality
}

fragment SchemaCompareErrorFields on SchemaCompareError {
  message
}

fragment SchemaVersionFields on SchemaVersion {
  id
  date
  valid
  commit {
    id
    author
    service
    commit
    __typename
  }
}
```

and compare `data.schemaCompareToPrevious.diff.before` and `data.schemaCompareToPrevious.diff.after` from response.

**Comment by TuvalSimha** [comment]
Hi @FluorescentHallucinogen , 
Thank you for your contribution. At the moment we are not looking into supporting this kind of feature in Inspector, because we think it might make maintenance more complicated, and since we think that visual diff is working fine, but it's not scalable after ~50 types. 

I do think this PR is a good reference for users who wants to create a custom setup on top of Voyager and Inspector, so maybe we can convert that into an example?

**Pull request closed without merging**
